security/pkg/nodeagent/cache/mock/secretcache_mock.go

```diff
@@ -131,3 +131,7 @@ func (s *TokenExchangeServer) ExchangeToken(context.Context, string, string) (st
 	// Since the secret cache uses the k8s token in the stored secret, we can just return anything here.
 	return "some-token", time.Now(), 200, nil
 }
+
+func (c *CAClient) Reconnect(isRotate bool) error {
```

Comment:
I don't think you need this method.

Reply:
This is an interface to close the existing connection and reset the connection. So that the connection will use the OUT_CERTS as the cert path otherwise by default it will use the PROV_CERT based on the isRotate flag. And as when we first start the citadel client we will use the PROV_CERT by setting the isRotate as false. I think this function is a way we transmit the connection to use the OUT_CERTS.